components/react/src/iconify.ts

```diff
@@ -247,7 +247,7 @@ class IconComponent extends React.Component<
 		super(props);
 		this.state = {
 			// Render placeholder before component is mounted
-			icon: { data: props.icon },
+			icon: window ? { data: props.icon } : null,
```

Comment:
That code won't work. It will throw an exception if `window` is not defined. You are checking value of an existing variable, not if a global variable exists. To check if a global variable exists, use `global.window` or wrap code that checks it in `try..catch`.

Reply:
Thanks, does it throw an exception because of the SSR?

Reply:
Because variable `window` does not exist in Node. It exists only in browser and only in window context (so not in service workers).

When you check for `window` or any other variable like that, you are checking value of existing variable, not if variable exists. If variable doesn't exist, JavaScript will throw an error. Open browser console, try typing something like `test123 ? true : false` and you'll get exception because variable does not exist.